app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt

```diff
@@ -44,35 +45,47 @@ class WallpaperSettingsFragment : Fragment() {
-                        loadWallpaperResource = {
-                            wallpaperManager.loadSavedWallpaper(requireContext(), it)
-                        },
-                        selectedWallpaper = currentWallpaper,
-                        onSelectWallpaper = { selectedWallpaper: Wallpaper ->
-                            currentWallpaper = selectedWallpaper
-                            wallpaperManager.currentWallpaper = selectedWallpaper
-                            metrics.track(Event.WallpaperSelected(selectedWallpaper))
-                        },
-                        onViewWallpaper = { findNavController().navigate(R.id.homeFragment) },
-                        tapLogoSwitchChecked = wallpapersSwitchedByLogo,
-                        onTapLogoSwitchCheckedChange = {
-                            settings.wallpapersSwitchedByLogoTap = it
-                            wallpapersSwitchedByLogo = it
-                            metrics.track(Event.ChangeWallpaperWithLogoToggled(it))
-                        }
-                    )
-                }
+            wallpaperSettings()
+        }
+    }
+
+    private fun ComposeView.wallpaperSettings() {
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+        setContent {
+            FirefoxTheme {
+                var currentWallpaper by remember { mutableStateOf(wallpaperManager.currentWallpaper) }
+                var wallpapersSwitchedByLogo by remember { mutableStateOf(settings.wallpapersSwitchedByLogoTap) }
+                WallpaperSettings(
+                    wallpapers = wallpaperManager.wallpapers,
+                    defaultWallpaper = WallpaperManager.defaultWallpaper,
+                    loadWallpaperResource = {
+                        wallpaperManager.loadSavedWallpaper(requireContext(), it)
+                    },
+                    selectedWallpaper = currentWallpaper,
+                    onSelectWallpaper = { selectedWallpaper: Wallpaper ->
+                        currentWallpaper = selectedWallpaper
+                        wallpaperManager.currentWallpaper = selectedWallpaper
+                        metrics.track(Event.WallpaperSelected(selectedWallpaper))
+                    },
+                    onViewWallpaper = { findNavController().navigate(R.id.homeFragment) },
+                    tapLogoSwitchChecked = wallpapersSwitchedByLogo,
+                    onTapLogoSwitchCheckedChange = {
+                        settings.wallpapersSwitchedByLogoTap = it
+                        wallpapersSwitchedByLogo = it
+                        metrics.track(Event.ChangeWallpaperWithLogoToggled(it))
+                    }
+                )
             }
         }
     }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        (view as? ComposeView)?.apply {
+            wallpaperSettings()
+        }
+    }
```

Comment:
Have you look at doing this with `DisposableEffect(LocalConfiguration.current.orientation) {}` which is more 'Compose'-y?